cpp/src/ray/runtime/task/task_executor.cc

```diff
@@ -79,10 +71,28 @@ std::pair<Status, std::shared_ptr<msgpack::sbuffer>> GetExecuteResult(
   }
 
   RAY_LOG(DEBUG) << "Get execute function" << func_name << " ok";
-  auto result = entry_func(func_name, args_buffer, actor_ptr);
-  RAY_LOG(DEBUG) << "Execute function" << func_name << " ok";
-  return std::make_pair(ray::Status::OK(),
-                        std::make_shared<msgpack::sbuffer>(std::move(result)));
+  try {
+    auto result = entry_func(func_name, args_buffer, actor_ptr);
+    RAY_LOG(DEBUG) << "Execute function" << func_name << " ok";
+    return std::make_pair(ray::Status::OK(),
+                          std::make_shared<msgpack::sbuffer>(std::move(result)));
+  } catch (ray::api::RayIntentionalSystemExitException &e) {
+    return std::make_pair(ray::Status::IntentionalSystemExit(), nullptr);
+  } catch (msgpack::type_error &e) {
+    return std::make_pair(
+        ray::Status::Invalid(std::string("invalid arguments: ") + e.what()), nullptr);
+  } catch (const std::invalid_argument &e) {
+    return std::make_pair(
+        ray::Status::Invalid(std::string("function execute exception: ") + e.what()),
+        nullptr);
+  } catch (const std::exception &e) {
+    return std::make_pair(
+        ray::Status::Interrupted(std::string("function execute exception: ") + e.what()),
+        nullptr);
+  } catch (...) {
+    return std::make_pair(ray::Status::UnknownError(std::string("unknown exception")),
+                          nullptr);
+  }
```

Comment:
What's the point of this catch?

Reply:
the user may not throw std::exception, maybe "throw error_no" , not throw a std::exception, we should also handle it.